CRT.c

```diff
@@ -724,3 +735,67 @@ void CRT_setColors(int colorScheme) {
 
    CRT_colors = CRT_colorSchemes[colorScheme];
 }
+
+void CRT_handleSIGSEGV(int signal) {
+   CRT_done();
+
+   fprintf(stderr, "\n\n"
+      "FATAL PROGRAM ERROR DETECTED\n"
+      "============================\n"
+      "Please report this issue at https://htop.dev including the following information:\n"
+      "\n"
+      "- Your htop version (htop --version)\n"
+      "- Your OS and kernel version (uname -a)\n"
+      "- Your distribution and release (lsb_release -a)\n"
+      "- Likely steps to reproduce (How did it happened?)\n"
+#ifdef HAVE_EXECINFO_H
+      "- Backtrace of the issue (see below)\n"
+#endif
+      "\n"
+   );
+
+   fprintf(stderr,
+      "Error information:\n"
+      "------------------\n"
+      "A signal %d was received.\n"
+      "\n",
+      signal
+   );
+
+#ifdef HAVE_EXECINFO_H
+   fprintf(stderr,
+      "Backtrace information:\n"
+      "----------------------\n"
+      "The following function calls were active when the issue was detected:\n"
+      "---\n"
+   );
+   size_t size = backtrace(backtraceArray, sizeof(backtraceArray) / sizeof(void *));
+   backtrace_symbols_fd(backtraceArray, size, 2);
+   fprintf(stderr,
+      "---\n"
+      "\n"
+      "To make the above information more practical to work with,\n"
+      "you should provide a disassembly of your binary.\n"
+      "This can usually be done by running the following command:\n"
+      "\n"
+      "   objdump -d -S -w `which htop` > ~/htop.objdump\n"
+      "\n"
+      "Please include the generated file in your report.\n"
+      "\n"
+   );
+#endif
+
+   fprintf(stderr,
+      "Running this program with debug symbols or inside a debugger may provide further insights.\n"
+      "\n"
+      "Thank you for helping to improve htop!\n"
+      "\n"
+      "htop " VERSION " aborting.\n"
+      "\n"
+   );
+
+   /* Call old sigsegv handler; may be default exit or third party one (e.g. ASAN) */
+   sigaction (signal, &old_sigsegv_handler, NULL);
+
+   kill(getpid(), signal);
```

Comment:
This can lead to infinite recursion in case the signal handler does not get reset.

Reply:
Okay, I'll add a check for that case.

Reply:
I was talking about the signals except SIGSEGV:
The SIGSEGV handler is installed with `SA_RESTART` and we are calling the old handler via `sigaction()`.
Installation through signal(2) might behave different on Linux and BSD.
Maybe just call `_exit`?